react-native/sample_integration/sample_integration/App.js

```diff
@@ -1,18 +1,35 @@
-import React, {useState} from 'react';
-import {View, StyleSheet, FlatList, Alert, SafeAreaView} from 'react-native';
+import React, {useState, useEffect} from 'react';
+import {StyleSheet, FlatList, Alert, SafeAreaView} from 'react-native';
 
 import Header from './components/Header';
 import ListItem from './components/ListItem';
 import AddItem from './components/AddItem';
-import {
-  startMoment,
-  endMoment,
-  logMessage,
-  INFO,
-  logBreadcrumb,
-} from 'react-native-embrace';
+import {startMoment, endMoment, logBreadcrumb} from 'react-native-embrace';
+import useConstructor from './util/custom_hooks';
 
 const App = () => {
+  // Flag true if user is currently editing an item
+  const [editStatus, editStatusChange] = useState(false);
+  // State to capture information about the item being edited
+  const [editItemDetail, editItemDetailChange] = useState({
+    id: null,
+    text: null,
+  });
+  const [checkedItems, checkedItemChange] = useState([]);
+
+  // EMBRACE HINT:
+  // Storing constants like moment or breadcrumb names make typos less likely.
+  const mountedMoment = 'App Mounted';
+
+  useConstructor(() => {
+    //This only happens ONCE and it happens BEFORE the initial render.
+
+    // EMBRACE HINT:
+    // Moments are a great way to measure the performance and abandonment of workflows within your application.
+    // We recommend to wrap non user interruptable flows with moments. For example, here we are measuring how long
+    // it takes this component to render.
+    startMoment(mountedMoment);
```

Comment:
I like this use case more than the previous one. However, we should do this for a different component since we should be ending the startup moment here instead since this is the main app component. So the things to change are:

* Move startMoment(mountedMoment) to a different component
* Call endStartup() here and add comment explaining what the startup moment is and why we have to end it.